src/sentry/replays/scripts/test_e2e.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import uuid
+import zlib
+
+import msgpack
+import requests
+
+from sentry.utils import json
+
+logger = logging.getLogger()
+
+
+def produce_replay_video_envelope(dsn: str, replay_video: bytes) -> None:
+    parts = dsn.split("/")
+    project_id = parts.pop()
+    dsn_url = parts.join("/") + f"/api/{project_id}/envelope/"
+
+    replay_id = uuid.uuid4().hex
+    logger.info("Replay envelope emitted with ID: %s", replay_id)
+
+    # Replay Envelope Headers
+    headers = {"event_id": replay_id, "dsn": dsn}
+
+    # Replay event envelope item.
+    replay_event = json.dumps(
+        {
+            "event_id": replay_id,
+            "replay_id": replay_id,
+            "segment_id": 0,
+            "replay_type": "session",
+            "error_sample_rate": 1,
+            "session_sample_rate": 1,
+            "timestamp": 1709314376,
+            "replay_start_timestamp": 1709314376 - 30,
+            "urls": [],
+            "error_ids": [],
+            "trace_ids": [],
+            "platform": "native",
+            "release": "release",
+            "dist": "mydist",
+            "environment": "production",
+            "tags": [],
+            # Important to send the SDK.  Apparently the client doesn't know how to act
+            # if you don't!
+            "sdk": {
+                "name": "sentry.javascript.react",
+                "version": "7.47.0",
+            },
+            "user": {
+                "id": "123",
+                "username": "Replay Video Test!",
+                "email": "replay-video@test.com",
+                "ip_address": "0.0.0.0",
+            },
+        }
+    ).encode()
+
+    # Replay recording envelope item.
+    replay_recording = zlib.compress(
```

Comment:
Maybe "replay_video_metadata" for this and "replay_video_bytes/data" for the input? Might be a little more clear